### PR TITLE
Add Cross-Origin Resource Sharing rules

### PR DESCRIPTION
**Pull request by oyelekci**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -34,6 +34,31 @@ variable "acl" {
...
 }
 
+variable "cors_allowed_headers" {
+  description = "Specifies which headers are allowed."
+  default     = ["Authorization"]
+}
+
+variable "cors_allowed_methods" {
+  description = "Specifies which methods are allowed. Can be GET, PUT, POST, DELETE or HEAD."
+  default     = ["GET"]
```
Comment (KashifSaadat) [inline variables.tf:44]: Are `Authorization` & `GET` default values for when no `cors_rule` is specified? i.e. not specifying any cors options would not change the spec of existing S3 buckets.

File: main.tf [hunk 2]
```diff
@@ -89,6 +89,14 @@ resource "aws_s3_bucket" "s3_bucket" {
   bucket = "${var.name}"
   acl    = "${var.acl}"
 
+  cors_rule {
```
Comment (KashifSaadat) [inline main.tf:92]: You may also want to include this block here: https://github.com/UKHomeOffice/acp-tf-s3/blob/v0.1.8/main.tf#L134

There's multiple `aws_s3_bucket` definitions with a count depending on what options are passed in.

Comment (oyelekci) [reply main.tf:92]: Good point, added

File: variables.tf (hunk 1 above)
Comment (oyelekci) [reply variables.tf:44]: As discussed and tested, these defaults are required and a permissive cors rule will get applied to each bucket that gets created

**Review by KashifSaadat — APPROVED** [review]
LGTM

**Pull request merged**